src/bio/bio_buffer.c

```diff
@@ -1047,7 +1047,7 @@ bio_memcpy(struct bio_desc *biod, uint16_t media, void *media_addr,
 		 * drain controller, however, test shows calling a persistent
 		 * copy and drain controller here is faster.
 		 */
-		if (DAOS_ON_VALGRIND && umem_tx_inprogress()) {
+		if (DAOS_ON_VALGRIND && umem_tx_inprogress(umem)) {
```

Comment:
I think we need to remove the DAOS_ON_VALGRIND check from these.   For metadata on SSD, we actually need to know the data is being modified even if no undo logging is needed.  I guess I can go through these and submit a PR.

Reply:
ok, I'll leave it to you then, I don't know much details of VALGRIND requirement TBH, thanks.